lms/djangoapps/teams/static/teams/js/views/team_profile.js

```diff
@@ -55,7 +55,8 @@
                         })
                     );
                     this.discussionView = new TeamDiscussionView({
-                        el: this.$('.discussion-module')
+                        el: this.$('.discussion-module'),
+                        readOnly: this.$('.discussion-module').data('read-only')
```

Comment:
Depending on how the value gets set on this data prop, `$.data()` can return a boolean.

https://jsfiddle.net/cmn0398n/

But yeah, it's probably safest to coerce it to a boolean here just to be safe.